File: Data Structures & Algorithms/valid-perfect-square/submission-0.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        l, r = 0, num
        while l <= r:
            m = (l + r) // 2
            res = m * m
            if res == num:
                return True
            if res > num:
                r = m - 1
            else:
                l = m + 1
        return False

File: Data Structures & Algorithms/majority-element/submission-0.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count = HashMap::new();

        for num in nums {
            *count.entry(num).or_insert(0) += 1;
        }

        *count.iter().max_by_key(|val| val.1).unwrap().0
    }
}
